-----------------------------------------------------------------
Revision: c4a874926d07b8597db4b78a181a89cf720a8418
Ancestor: dcd1194eb2bd80a296737c518ce2474661d9fa1f
Author: darkrain42@pidgin.im
Date: 05/16/2010 04:50:23 PM
Branch: im.pidgin.pidgin

Modified files:
        ChangeLog pidgin/gtkdocklet-gtk.c

ChangeLog: 

pidgin: Add charkins's patch for gtkstatusicon blinking.  Fixes #11855

============================================================
--- pidgin/gtkdocklet-gtk.c	621886961dfcbcd728e875983d4e131264731683
+++ pidgin/gtkdocklet-gtk.c	4c8694f749bba34ea1eda230978b1173e7353de6
@@ -75,6 +75,12 @@ docklet_gtk_status_update_icon(PurpleSta
 	if (icon_name) {
 		gtk_status_icon_set_from_icon_name(docklet, icon_name);
 	}
+
+	if (purple_prefs_get_bool(PIDGIN_PREFS_ROOT "/docklet/blink")) {
+		gtk_status_icon_set_blinking(docklet, (pending && !connecting));
+	} else if (gtk_status_icon_get_blinking(docklet)) {
+		gtk_status_icon_set_blinking(docklet, FALSE);
+	}
 }
 
 static void
